jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TraceContextCodec.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jaegertracing.internal.propagation;
+
+import io.jaegertracing.internal.JaegerObjectFactory;
+import io.jaegertracing.internal.JaegerSpanContext;
+import io.jaegertracing.spi.Codec;
+import io.opentracing.propagation.TextMap;
+import java.util.Collections;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Implementation of the TraceContext propagation protocol. See <a
+ * href=https://github.com/w3c/distributed-tracing>w3c/distributed-tracing</a>.
+ *
+ * This implementation is mostly copied over from OpenTelemetry Java SDK
+ * https://github.com/open-telemetry/opentelemetry-java/blob/ed98c35c0569a48f66339769913670334d6c8a95/api/src/main/java/io/opentelemetry/trace/propagation/HttpTraceContext.java#L40
+ */
+@Slf4j
+public class TraceContextCodec implements Codec<TextMap> {
+
+  static final String TRACE_PARENT = "traceparent";
+  static final String TRACE_STATE = "tracestate";
+
+  private static final String VERSION = "00";
+  private static final int VERSION_SIZE = 2;
+  private static final char TRACEPARENT_DELIMITER = '-';
+  private static final int TRACEPARENT_DELIMITER_SIZE = 1;
+  private static final int TRACE_ID_HEX_SIZE = 2 * 16;
+  private static final int SPAN_ID_HEX_SIZE = 2 * 8;
+  private static final int TRACE_OPTION_HEX_SIZE = 2;
```

Comment:
TRACE_OPTION is an odd name, can we use TRACE_FLAGS, which matches the w3c terminology?